src/Profiles.ts

```diff
@@ -119,12 +140,19 @@ export class Profiles {
     public async refresh(): Promise<void> {
         this.allProfiles = [];
         this.allTypes = [];
+        this.profilesForValidation = [];
         for (const type of ZoweExplorerApiRegister.getInstance().registeredApiTypes()) {
             const profileManager = await this.getCliProfileManager(type);
             const profilesForType = (await profileManager.loadAll()).filter((profile) => {
                 return profile.type === type;
             });
             if (profilesForType && profilesForType.length > 0) {
+                for (const validateProfile of profilesForType) {
```

Comment:
As discussed on Slack. We should avoid validating all profiles automatically as it is potentially a super expensive operation. Refresh is called during activation as well as every time an extender registers itself as well as every time the user clicks the refresh button. We do not want to run expensive network operations on all profiles every time. 

Especially not during activation as it would slow down the editor startup time, which is highly important for the user experience and in environments such as Che could even lead to timeouts.

An alternative would be to validate profiles when the user explicitly requests it or when the use a profile the first time such as doing the first Search query as well as after Create/Edit Profile.

Reply:
I believe this was addressed 😋